### PR TITLE
URL encode database connection URI components

### PR DESCRIPTION
**Pull request by jgpruitt**
Database passwords frequently contain characters that must
be URL encoded when put in the Postgres connection URI. If
not, the URI is invalid and will not work. This addresses
bug report in #199.

### CODE REVIEW

**Review by niksajakovljevic — APPROVED** [review]

**Review by JamesGuthrie — APPROVED** [review]

File: chart/Chart.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ type: application
 
 version: 0.7.0
 appVersion: 0.7.0
-kubeVersion: ">= 1.19.0 < 1.22.0"
+kubeVersion: ">= 1.19.0 <= 1.22.0"
```
Comment (jgpruitt) [inline chart/Chart.yaml:9]: @VineethReddy02 Is this change okay with you? I'm not sure if we had it limited to < 1.22.0 for a specific reason.

Comment (VineethReddy02) [reply chart/Chart.yaml:9]: Yup, the otel-operator doesn't support 1.22 version on uninstall flow leaves the child components. So we decided not to support 1.22 version atm.

File: chart/Chart.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@ type: application
 
 version: 0.7.0
 appVersion: 0.7.0
-kubeVersion: ">= 1.19.0 < 1.22.0"
+kubeVersion: ">= 1.19.0-0 < 1.22.0-0"
```
Comment (paulfantom) [inline chart/Chart.yaml:9]: Why `-0` is needed?

File: .github/workflows/generate.yml [hunk 3]
```diff
@@ -23,6 +23,8 @@ jobs:
 
       - name: Test static generation
         run: |
+          curl -LO https://dl.k8s.io/release/v1.21.0/bin/linux/amd64/kubectl
+          sudo install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl
```
Comment (paulfantom) [inline .github/workflows/generate.yml:27]: I don't see any added dependency on kubectl in this PR. Why is this part needed?

File: chart/Chart.yaml (hunk 2 above)
Comment (VineethReddy02) [reply chart/Chart.yaml:9]: These are just added to test is kubectl, Kubernetes server versions in CI are causing the issue. They are not required reverting back my changes.

File: .github/workflows/generate.yml (hunk 3 above)
Comment (VineethReddy02) [reply .github/workflows/generate.yml:27]: Same as the above reason.

**Comment by VineethReddy02** [comment]
Dependent on #202 (CI passes once the 202 gets in)
